neuro_san/client/direct_agent_session_factory.py

```diff
@@ -24,6 +24,7 @@
 from neuro_san.internals.graph.persistence.registry_manifest_restorer import RegistryManifestRestorer
 from neuro_san.internals.interfaces.agent_network_provider import AgentNetworkProvider
 from neuro_san.internals.network_providers.agent_network_storage import AgentNetworkStorage
+from neuro_san.service.generic.asyncio_executor_pool import AsyncioExecutorPool
```

Comment:
Cannot import stuff from neuro_san.service in the client area.  Huge reach-around.
I think what you want to do is put this in internals.utils for now.